inst/app/modules/main_map.R

```diff
@@ -9,12 +27,45 @@ hk_accidents_valid_sf <- st_as_sf(x = hk_accidents_valid, coords = c("longitude"
 # TODO: investigate why
 hk_accidents_valid_sf$Date <- as.Date(hk_accidents_valid_sf$Date, format = "%Y-%m-%d")
 
-CARTODB_POSITRON_TILE_URL <- "https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png"
+# Combine date and time together as a complete POSIXct class time column
+# Easier for formatting
+hk_accidents_valid_sf$Date_Time <- as.POSIXct(
+  strptime(
+    paste0(hk_accidents_valid_sf$Date, " ", hk_accidents_valid_sf$Time),
+    format = "%Y-%m-%d %H%M",
+    tz = "Asia/Hong_Kong"
+    )
+  )
+
 
 output$main_map <- renderLeaflet({
-  leaflet() %>%
-    addTiles(urlTemplate = CARTODB_POSITRON_TILE_URL) %>%
-    setView(lng = 114.2, lat = 22.3, zoom = 12)
+  overview_map <- leaflet() %>%
+    setView(lng = 114.2, lat = 22.3, zoom = 12) %>%
+    # Add geocoder map widget
+    addSearchOSM(options = searchOptions(hideMarkerOnCollapse = TRUE))
+
+  # Subset basemap providers to be used for the map
+  SELECTED_BASEMAPS <- leaflet::providers[c("CartoDB.Positron", "Stamen.TonerLite", "OpenStreetMap")]
+
+  # Add the basemap tiles in background
+  # http://rstudio.github.io/leaflet/morefeatures.html
+  for (provider in SELECTED_BASEMAPS) {
+    overview_map <- addProviderTiles(overview_map, provider, group = provider)
+  }
+
+  # Add change basemap widget
+  overview_map %>%
+    addLayersControl(baseGroups = names(basemap_providers),
```

Comment:
I'm getting an error here, and I think it's because there does not seem to be any other code which initialises `basemap_providers`. 

This is the screenshot of the Shiny app.
![image](https://user-images.githubusercontent.com/25527485/122358604-b3722b00-cf4c-11eb-92c7-ca84e30eba4f.png)

This is the output in the console:
```
Listening on http://127.0.0.1:4050
`summarise()` ungrouping output (override with `.groups` argument)
`summarise()` ungrouping output (override with `.groups` argument)
Warning: Error in evalAll: object 'basemap_providers' not found
  106: evalAll
  105: evalFormula
  104: invokeMethod
  103: addLayersControl
  102: function_list[[i]]
  101: freduce
  100: _fseq
   99: eval
   98: eval
   96: %>%
   95: func [modules/main_map.R#57]
   82: origRenderFunc
   81: output$main_map
    1: runApp
Warning: Ignoring unknown parameters: binwidth, bins, **pad**
```

Reply:
My bad. `basemap_providers` should be `SELECTED_BASEMAPS`. Thanks for spotting this out. 🤦🏻